helix-term/src/ui/picker.rs

```diff
@@ -690,20 +690,19 @@ impl<T: Item + 'static> Picker<T> {
                 }
             };
 
-            // align to middle
-            let first_line = range
-                .map(|(start, end)| {
-                    let height = end.saturating_sub(start) + 1;
-                    let middle = start + (height.saturating_sub(1) / 2);
-                    middle.saturating_sub(inner.height as usize / 2).min(start)
-                })
-                .unwrap_or(0);
-
-            let offset = ViewPosition {
-                anchor: doc.text().line_to_char(first_line),
-                horizontal_offset: 0,
-                vertical_offset: 0,
-            };
+            let text_fmt = doc.text_format(inner.width, None);
+            let annotations = TextAnnotations::default();
+            let mut offset = ViewPosition::default();
+            (offset.anchor, offset.vertical_offset) = char_idx_at_visual_offset(
```

Comment:
If `range` is `None` we don't need need this. Since scrolling upwards from 0 is not possible anyway so I would siimoly change this to 

``` rust
if let Some(range) = range {
 (offset.anchor, offset.vertical_offset) = ...
}
```